backend/src/nodes/impl/onnx/session.py

```diff
@@ -4,14 +4,21 @@
 
 import onnxruntime as ort
 
+from system import is_arm_mac
+
 from ...utils.exec_options import ExecutionOptions
 from .model import OnnxModel
 
 
 def create_inference_session(
     model: OnnxModel, exec_options: ExecutionOptions
 ) -> ort.InferenceSession:
-    if exec_options.onnx_execution_provider == "TensorrtExecutionProvider":
+    if is_arm_mac:
+        providers = [
+            "CoreMLExecutionProvider",
+            "CPUExecutionProvider",
+        ]
```

Comment:
So we're just going to ignore execution options?

Reply:
Oh hmm.. I guess I need to add it as an option there